Make the iSCSI page compliant with the diataxis framework

network-charles opened the pull request:
### Description

This PR makes the iSCSI page compliant with the diataxis framework

---

### Related Issue

- Fixes: #228

---

### Checklist

- [x] I have read and followed the [Ubuntu Server contributing guide](https://documentation.ubuntu.com/server/contributing/).
- [x] I have signed the [Contributor License Agreement (CLA)](https://ubuntu.com/legal/contributors).
- [x] My pull request is linked to an existing issue (if applicable).
- [x] I have tested my changes, and they work as expected.

s-makin commented on redirects.txt:
```diff
@@ -438,8 +438,8 @@ troubleshooting-tls-ssl explanation/crypto/troubleshooting-tls-ssl/
 
-service-iscsi explanation/storage/iscsi-initiator-or-client/
-iscsi-initiator-or-client explanation/storage/iscsi-initiator-or-client/
+service-iscsi how-to/storage/iscsi-initiator-or-client/
+iscsi-initiator-or-client how-to/storage/iscsi-initiator-or-client/
 
 about-apt-upgrade-and-phased-updates explanation/software/about-apt-upgrade-and-phased-updates/
```
```suggestion
service-iscsi how-to/storage/iscsi-initiator-or-client/
iscsi-initiator-or-client how-to/storage/iscsi-initiator-or-client/
explanation/storage/iscsi-initiator-or-client/ how-to/storage/iscsi-initiator-or-client/

about-apt-upgrade-and-phased-updates explanation/software/about-apt-upgrade-and-phased-updates/
```

network-charles replied:
Nice! Thank you

s-makin reviewed (approved):
Thanks for making this change! Everything is redirecting correctly in the preview so all the bases are covered. I am getting a warning though about a merge conflict. If you wouldn't mind resolving that, then I'll be very happy to merge :)

network-charles commented:
I can't see a merge conflict locally. The page was moved from `explanation/storage/iscsi-initiator-or-client.md` to `how-to/storage/iscsi-initiator-or-client.md`.

I wonder if it's the new redirect addition at Line 443:
`explanation/storage/iscsi-initiator-or-client/ how-to/storage/iscsi-initiator-or-client/`

I'll temporarily remove it and see if it fixes the pipeline.

network-charles commented:
Oh, I see it now. A new change was added to `explanation/storage/iscsi-initiator-or-client.md` on the main branch.

network-charles commented:
It didn't fix it.
@s-makin, can you kindly take a look? I don't know why it's not working

network-charles commented:
I just did a git checkout from this branch to a new one and tried to rebase from the `main` branch, then I saw the conflict.

```bash
➜   git checkout -b patch-2 
➜   git status
interactive rebase in progress; onto b1956f2
Last command done (1 command done):
   pick 2ee0225 make the iscsi page compliant with the diataxis framework
Next commands to do (3 remaining commands):
   pick a89c445 Suggestion 1
   pick e3906d8 remove new redirect and check pipeline
  (use "git rebase --edit-todo" to view and edit)
You are currently rebasing branch 'patch-2' on 'b1956f2'.
  (fix conflicts and then run "git rebase --continue")
  (use "git rebase --skip" to skip this patch)
  (use "git rebase --abort" to check out the original branch)

Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        modified:   explanation/storage.rst
        deleted:    explanation/storage/iscsi-initiator-or-client.md
        modified:   how-to/storage.rst
        modified:   redirects.txt

Unmerged paths:
  (use "git restore --staged <file>..." to unstage)
  (use "git add <file>..." to mark resolution)
        both modified:   how-to/storage/iscsi-initiator-or-client.md
```

I will do a rebase now to fix the conflict and then push the changes.

network-charles commented:
Wow, this was very tricky. I’m glad it's resolved now.

Next time when I experience this issue, I’ll do a rebase and pay attention to the conflicted file on my editor.

s-makin commented:
> Wow, this was very tricky. I’m glad it's resolved now.
> 
> Next time when I experience this issue, I’ll do a rebase and pay attention to the conflicted file on my editor.

You were too quick for me! Good job on fixing the issue before I even got a chance to look it over :)
